### ISSUE TITLE
json callback plugin fails on JSON encoding with in-line vault variables

### ISSUE THREAD

**Issue by AlanCoding** [ansible/ansible#46957]
##### SUMMARY
Using the json callback plugin results in an exception if in-line vaulted variables are included by a task.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
lib/ansible/plugins/callback/json.py

but others are probably affected too @bcoca may have more to contribute

##### ANSIBLE VERSION
```paste below
ansible 2.8.0.dev0
  config file = None
  configured module search path = [u'/Users/alancoding/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/alancoding/.virtualenvs/ansible/lib/python2.7/site-packages/ansible-2.8.0.dev0-py2.7.egg/ansible
  executable location = /Users/alancoding/.virtualenvs/ansible/bin/ansible
  python version = 2.7.11 (default, Oct 17 2016, 14:59:40) [GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
```

##### CONFIGURATION
```paste below
ANSIBLE_NOCOWS(env: ANSIBLE_NOCOWS) = True
```

##### OS / ENVIRONMENT
N/A, local action, Mac OS

##### STEPS TO REPRODUCE
Steps are stored in source control at https://github.com/AlanCoding/utility-playbooks/tree/master/include_vars

playbook `vault_vars.yml`, bug is in "Include contents from single-variable encrypted vars"

```yaml
- name: Include vault vars
  hosts: localhost
  connection: local
  gather_facts: False

  tasks:

    - name: Include contents from whole-file encrypted vars
      include_vars: stuff.yml

    - name: debug var from whole-file encrypted stuff
      debug: var=spoiler_text

    - name: Include contents from single-variable encrypted vars
      include_vars: thing.yml

    - name: debug var from single var encrypted value
      debug: var=spoiler_text
```

run via CLI:

```
ANSIBLE_STDOUT_CALLBACK=json ansible-playbook -i localhost, include_vars/vault_vars.yml --vault-id=alan@prompt -vvv
```

provide vault password "password"

##### EXPECTED RESULTS
JSON serialization of playbook run

##### ACTUAL RESULTS

```paste below
ANSIBLE_STDOUT_CALLBACK=json ansible-playbook -i localhost, include_vars/vault_vars.yml --vault-id=alan@prompt -vvv
ansible-playbook 2.8.0.dev0
  config file = None
  configured module search path = [u'/Users/alancoding/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/alancoding/.virtualenvs/ansible/lib/python2.7/site-packages/ansible-2.8.0.dev0-py2.7.egg/ansible
  executable location = /Users/alancoding/.virtualenvs/ansible/bin/ansible-playbook
  python version = 2.7.11 (default, Oct 17 2016, 14:59:40) [GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
No config file found; using defaults
Vault password (alan): 
Parsed localhost, inventory source with host_list plugin
1 plays in include_vars/vault_vars.yml
META: ran handlers
META: ran handlers
META: ran handlers
 [WARNING]: Failure using method (v2_playbook_on_stats) in callback plugin (<ansible.plugins.callback.json.CallbackModule object at 0x1114b9890>): u'people run into some space
aliens and they end up fighting them' is not JSON serializable

Callback Exception: 
  File "/Users/alancoding/.virtualenvs/ansible/lib/python2.7/site-packages/ansible-2.8.0.dev0-py2.7.egg/ansible/executor/task_queue_manager.py", line 375, in send_callback
    method(*new_args, **kwargs)
   File "/Users/alancoding/.virtualenvs/ansible/lib/python2.7/site-packages/ansible-2.8.0.dev0-py2.7.egg/ansible/plugins/callback/json.py", line 117, in v2_playbook_on_stats
    self._display.display(json.dumps(output, indent=4, sort_keys=True))
   File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/__init__.py", line 251, in dumps
    sort_keys=sort_keys, **kw).encode(obj)
   File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 209, in encode
    chunks = list(chunks)
   File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 434, in _iterencode
    for chunk in _iterencode_dict(o, _current_indent_level):
   File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
   File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 332, in _iterencode_list
    for chunk in chunks:
   File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
   File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 332, in _iterencode_list
    for chunk in chunks:
   File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
   File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
   File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
   File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
   File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 442, in _iterencode
    o = _default(o)
   File "/usr/local/Cellar/python/2.7.11/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 184, in default
    raise TypeError(repr(o) + " is not JSON serializable")
```

**Comment by bcoca** [comment]
fixed via #46830, which does include other callbacks with the issue

**Comment by AlanCoding** [comment]
Re-tested with current `devel`, results:

```
ANSIBLE_STDOUT_CALLBACK=json ansible-playbook -i localhost, include_vars/vault_vars.yml --vault-id=alan@prompt -vvv
ansible-playbook 2.8.0.dev0
  config file = None
  configured module search path = [u'/Users/alancoding/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/alancoding/.virtualenvs/ansible/lib/python2.7/site-packages/ansible-2.8.0.dev0-py2.7.egg/ansible
  executable location = /Users/alancoding/.virtualenvs/ansible/bin/ansible-playbook
  python version = 2.7.11 (default, Oct 17 2016, 14:59:40) [GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
No config file found; using defaults
Vault password (alan): 
Parsed localhost, inventory source with host_list plugin
1 plays in include_vars/vault_vars.yml
META: ran handlers
META: ran handlers
META: ran handlers
{
    "custom_stats": {}, 
    "global_custom_stats": {}, 
    "plays": [
        {
            "play": {
                "duration": {
                    "end": "2018-10-12T13:50:32.560377Z", 
                    "start": "2018-10-12T13:50:32.221447Z"
                }, 
                "id": "a0999b12-fe9b-9302-9922-000000000006", 
                "name": "Include vault vars"
            }, 
            "tasks": [
                {
                    "hosts": {
                        "localhost": {
                            "_ansible_no_log": false, 
                            "action": "include_vars", 
                            "ansible_facts": {
                                "spoiler_text": "people run into some space aliens\nand they end up fighting them"
                            }, 
                            "ansible_included_var_files": [
                                "/Users/alancoding/Documents/repos/utility-playbooks/include_vars/stuff.yml"
                            ], 
                            "changed": false
                        }
                    }, 
                    "task": {
                        "duration": {
                            "end": "2018-10-12T13:50:32.303758Z", 
                            "start": "2018-10-12T13:50:32.231191Z"
                        }, 
                        "id": "a0999b12-fe9b-9302-9922-000000000008", 
                        "name": "Include contents from whole-file encrypted vars"
                    }
                }, 
                {
                    "hosts": {
                        "localhost": {
                            "_ansible_no_log": false, 
                            "_ansible_verbose_always": true, 
                            "action": "debug", 
                            "changed": false, 
                            "spoiler_text": "people run into some space aliens\nand they end up fighting them"
                        }
                    }, 
                    "task": {
                        "duration": {
                            "end": "2018-10-12T13:50:32.356774Z", 
                            "start": "2018-10-12T13:50:32.312589Z"
                        }, 
                        "id": "a0999b12-fe9b-9302-9922-000000000009", 
                        "name": "debug var from whole-file encrypted stuff"
                    }
                }, 
                {
                    "hosts": {
                        "localhost": {
                            "_ansible_no_log": false, 
                            "action": "include_vars", 
                            "ansible_facts": {
                                "spoiler_text": {
                                    "__ansible_vault": "$ANSIBLE_VAULT;1.2;AES256;alan\n36386366323637343336383236333964643463626232663931323932616365616263646434356166\n6533313834316664626239396336326333643163353433340a623537336431313534393531663464\n65316639396361316664373732366438636638336439636166643832626432646231313633396666\n6435326436663766370a663466633064376161336439303166353139326335313665363030363437\n66643364343863366365343964343764316231373638633633303265363763613836313435313266\n31326665373432376461623162353038643935346633643862646632346532396263356135376431\n613262626339313336643338653330613164\n"
                                }
                            }, 
                            "ansible_included_var_files": [
                                "/Users/alancoding/Documents/repos/utility-playbooks/include_vars/thing.yml"
                            ], 
                            "changed": false
                        }
                    }, 
                    "task": {
                        "duration": {
                            "end": "2018-10-12T13:50:32.409353Z", 
                            "start": "2018-10-12T13:50:32.365541Z"
                        }, 
                        "id": "a0999b12-fe9b-9302-9922-00000000000a", 
                        "name": "Include contents from single-variable encrypted vars"
                    }
                }, 
                {
                    "hosts": {
                        "localhost": {
                            "_ansible_no_log": false, 
                            "_ansible_verbose_always": true, 
                            "action": "debug", 
                            "changed": false, 
                            "spoiler_text": "people run into some space aliens and they end up fighting them"
                        }
                    }, 
                    "task": {
                        "duration": {
                            "end": "2018-10-12T13:50:32.488974Z", 
                            "start": "2018-10-12T13:50:32.416068Z"
                        }, 
                        "id": "a0999b12-fe9b-9302-9922-00000000000b", 
                        "name": "debug var from single var encrypted value"
                    }
                }, 
                {
                    "hosts": {
                        "localhost": {
                            "_ansible_no_log": false, 
                            "action": "set_fact", 
                            "ansible_facts": {
                                "bar": "people run into some space aliens and they end up fighting them"
                            }, 
                            "changed": false
                        }
                    }, 
                    "task": {
                        "duration": {
                            "end": "2018-10-12T13:50:32.560377Z", 
                            "start": "2018-10-12T13:50:32.494245Z"
                        }, 
                        "id": "a0999b12-fe9b-9302-9922-00000000000c", 
                        "name": "set fact with spoiler_text"
                    }
                }
            ]
        }
    ], 
    "stats": {
        "localhost": {
            "changed": 0, 
            "failures": 0, 
            "ok": 5, 
            "skipped": 0, 
            "unreachable": 0
        }
    }
}
```
